compiler/src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -140,14 +141,14 @@ object Settings {
           else if (!choices.contains(argRest))
             fail(s"$arg is not a valid choice for $name", args)
           else update(argRest, args)
-        case (StringTag, arg :: args) if name == "-d" =>
-          Path(arg) match {
-            case _: Directory =>
-              update(arg, args)
-            case p if p.extension == "jar" =>
-              update(arg, args)
-            case _ =>
-              fail(s"'$arg' does not exist or is not a directory", args)
+        case (OutputTag, arg :: args) =>
+          val path = Directory(arg)
+          val isJar = path.extension == "jar"
+          if (!isJar && !path.isDirectory)
+            fail(s"'$arg' does not exist or is not a directory or .jar file", args)
+          else {
+            val output = if (isJar) JarArchive.create(path) else new PlainDirectory(path)
```

Comment:
We will now hold on to an open jar during the whole compiler run instead of just backend. We didn't even open the jar before if we stopped before backend

Reply:
It's an empty open jar, so not a big deal I think (it cannot be deleted on Windows while it's open, but who cares?)